### PR TITLE
Use JsonSerializable, update score logic

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by kevmoo** [comment]
@isoos – I wanted to pull the scores out into a JSON file so it's easy to update them from a script. While I was at it, I converted the rest of the JSON logic to JsonSerializable.

File: app/lib/shared/search_service.g.dart [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        url: json['url'] as String,
+        package: json['package'] as String,
+        version: json['version'] as String,
+        devVersion: json['devVersion'] as String,
+        description: json['description'] as String,
+        lastUpdated: json['lastUpdated'] as String,
+        readme: json['readme'] as String,
+        detectedTypes:
+            (json['detectedTypes'] as List)?.map((e) => e as String)?.toList(),
+        popularity: json['popularity'] as double);
```
Comment (mkustermann) [inline app/lib/shared/search_service.g.dart]: Side note: I'm not quite sure if the generator is doing this correctly: Often the wireformat json contains numbers as either as doubles or integers (if there is no fractional part). A `0 as double` will throw an exception!

So in order to work around this, it's better if the generator would generate: `(json['popularity'] as num).toDouble()` here. This ensures that the resulting value is a double, even if it was encoded as a whole integer.

File: app/lib/shared/mock_scores.dart [hunk 2]
```diff
@@ -1,863 +1,9 @@
+library pub_dartlang_org.shared.mock_scores;
```
Comment (mkustermann) [inline app/lib/shared/mock_scores.dart]: Theoretically there should be a copy right notice here!

File: app/lib/shared/mock_scores.json [hunk 3]
```diff
@@ -0,0 +1,1182 @@
...
+ "bbs": 0,
+ "preprocessor": 0,
+ "highcharts_options": 0,
+ "cupid": 0,
+ "jaguar_facebook_client": 0,
+ "multi_reg_exp": 0,
+ "newton": 0,
+ "sync_socket": 0,
+ "hub": 0,
+ "tentacle_response_formatter": 0
```
Comment (mkustermann) [inline app/lib/shared/mock_scores.json]: Here, e.g., we have a json file with either integers or doubles!

File: app/lib/shared/search_service.g.dart (hunk 1 above)
Comment (kevmoo) [inline app/lib/shared/search_service.g.dart]: Great catch!

File: app/lib/shared/mock_scores.json (hunk 3 above)
Comment (kevmoo) [inline app/lib/shared/mock_scores.json]: On second thought, I think dart2js handles (`0 as double`) quite fine.

Comment (kevmoo) [inline app/lib/shared/mock_scores.json]: ...in fact, it must – because I"ve used this code a great deal on web apps.

**Comment by kevmoo** [comment]
@isoos – thoughts?

**Comment by isoos** [comment]
lgtm pending the instructions in readme.md

File: app/lib/shared/mock_scores.json (hunk 3 above)
Comment (mkustermann) [inline app/lib/shared/mock_scores.json]: We're not running on dart2js but rather on the VM, which has the proper number semantics.

```
 % cat test.dart
main() => print(1 as double);

 % dart test.dart
Unhandled exception:
type 'int' is not a subtype of type 'double' in type cast where
  int is from dart:core
  double is from dart:core

#0      Object._as (dart:core-patch/object_patch.dart:73)
#1      main (file:///usr/local/google/home/kustermann/repositories/vipunen-sdk/sdk/test.dart:1:19)
#2      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:265)
#3      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:151)
```

Comment (kevmoo) [inline app/lib/shared/mock_scores.json]: I just realized this and verified. Sorry....digging...

File: app/lib/shared/search_service.g.dart (hunk 1 above)
Comment (kevmoo) [inline app/lib/shared/search_service.g.dart]: Fixed

**Comment by kevmoo** [comment]
README work coming...

File: app/lib/shared/mock_scores.dart (hunk 2 above)
Comment (kevmoo) [inline app/lib/shared/mock_scores.dart]: ack

**Pull request merged**